python/ray/data/dataset.py

```diff
@@ -5059,7 +5059,18 @@ def __setstate__(self, state):
         self._current_executor = None
 
     def __del__(self):
-        if self._current_executor and ray is not None and ray.is_initialized():
+        if not self._current_executor:
+            return
+
+        # When Python shuts down, `ray` might evaluate to `<module None from None>`.
+        # This value is truthy and not `None`, so we use a try-catch rather than
+        # something like `if ray is not None`. For more information, see #42382.
+        try:
+            is_ray_initialized = ray.is_initialized()
```

Comment:
Nit: Do we need to store the value i.e. would just the following work:
```
try:
    if ray.is_initialized():
        self._current_executor.shutdown()
except TypeError:
    pass
```

Reply:
Nah, that'd totally work. 

Updated..